### PR TITLE
Okx: Various websocket fixes

### PR DESCRIPTION
**Pull request by gbjk**
Bundle of Okx fixes for websockets.
Commit messages should provide adequate context.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

### CODE REVIEW

File: exchanges/okx/okx_websocket.go [hunk 1]
```diff
@@ -385,10 +386,25 @@ func (ok *Okx) handleSubscription(operation string, subscriptions []stream.Chann
 		var algoID string
 		var uid string
 
-		if arg.Channel == okxChannelAccount ||
-			arg.Channel == okxChannelOrders {
+		switch arg.Channel {
```
Comment (shazbert) [inline exchanges/okx/okx_websocket.go:389]: 🔥

File: exchanges/okx/okx_types.go [hunk 2]
```diff
@@ -1670,30 +1670,30 @@ type RFQOrderLeg struct {
...
 }
 
 // CancelRFQResponse represents cancel RFQ orders response
 type CancelRFQResponse struct {
-	RfqID               string `json:"rfqId"`
-	ClientSuppliedRfqID string `json:"clRfqId"`
-	StatusCode          string `json:"sCode"`
-	StatusMsg           string `json:"sMsg"`
+	RfqID       string `json:"rfqId"`
+	ClientRfqID string `json:"clRfqId"`
```
Comment (shazbert) [inline exchanges/okx/okx_types.go:1694]: I am being pedantic but some are `ClientRfqID` and `ClientRFQID` could we please standardise?

**Review by shazbert — CHANGES_REQUESTED** [review]
Pretty much tACK. One pedantic nit.

File: exchanges/okx/okx_types.go (hunk 2 above)
Comment (gbjk) [reply exchanges/okx/okx_types.go:1694]: Well... You see ... If we're going to be Pedantic about it then prepositions should be lowercase ...
So RfQId 😂

Comment (gbjk) [reply exchanges/okx/okx_types.go:1694]: I'll go for RfqID
I tried all three and it feels best.

This highlights that 1688 is already wrong too. That's an array.

Comment (gbjk) [reply exchanges/okx/okx_types.go:1694]: Fixed in 3019f93b

File: exchanges/okx/okx_websocket.go (hunk 1 above)
Comment (gbjk) [reply exchanges/okx/okx_websocket.go:389]: What I noticed during this is that switching implies type conversion.
You can switch an `any` and it'll cast and compare.

**Review by shazbert — APPROVED** [review]
Changes ACK. TA

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice, thank you for this!

File: exchanges/okx/okx_test.go [hunk 3]
```diff
@@ -1905,7 +1905,7 @@ func TestUpdateOrderExecutionLimits(t *testing.T) {
 		}
 		stepIncr := 0.1
 		if a == asset.Options {
-			stepIncr = 0.0005
```
[the remark this answers is not in the document]
Comment (gbjk) [reply exchanges/okx/okx_test.go]: Completely agree.
I made a few stupid mistakes like this.
Fixed with 049eb219

File: exchanges/okx/okx_websocket.go [hunk 4]
```diff
@@ -1044,19 +1060,30 @@ func (ok *Okx) wsProcessOrders(respRaw []byte) error {
 		if err != nil {
 			return err
 		}
+		orderAmount := response.Data[x].Size
+		if response.Data[x].QuantityType == "quote_ccy" {
```
[the remark this answers is not in the document]
Comment (gbjk) [reply exchanges/okx/okx_websocket.go:1064]: Yeah, and now I don't have any reason we didn't populate QuoteAmount 🤷

[the remark this answers is not in the document]
Comment (gbjk) [reply exchanges/okx/okx_websocket.go:1066]: Fixed with 6048ac46

**Review by gloriousCode — APPROVED** [review]
tACK! Thank you!

**Comment by gbjk** [comment]
Rebasing 🔨

**Comment by gbjk** [comment]
@gloriousCode Rebased.

One thing that wasn't clean was the RFQ renaming, which wasn't very thorough the first time I did it.
Really wish we hadn't engaged in it on this PR now.

I've standardised *all* usages to Rfq now, including error messages and method names.

**Review by shazbert — APPROVED** [review]
tACK. 💯

**Review by thrasher- — APPROVED** [review]
Good stuff @gbjk !

**Pull request merged**